registry_module.go

```diff
@@ -690,6 +799,37 @@ func (o RegistryModuleID) valid() error {
 	return nil
 }
 
+func (o RegistryModuleDeleteByNameOrProvider) valid() error {
+	if !validStringID(&o.Organization) {
+		return ErrInvalidOrg
+	}
+
+	if !validString(&o.Name) {
+		return ErrRequiredName
+	}
+
+	if !validStringID(&o.Name) {
+		return ErrInvalidName
+	}
+
+	switch o.RegistryName {
+	case PublicRegistry:
+		if !validString(&o.Namespace) {
+			return ErrRequiredNamespace
+		}
+	case PrivateRegistry:
+		if &o.Namespace != &o.Organization {
+			return ErrOrgRequiredPrivateReqistry
+		}
+	case "":
+		// RegistryName is not required when deleting by name
```

Comment:
Should `RegistryName` be required? the query doesn't seem to work without the RegistryName

Reply:
Yep. Corrected!